python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -118,30 +52,45 @@ class MTAirCompressorCsc(salobj.BaseCsc):
     valid_simulation_modes: typing.Sequence[int] = (0, 1)
 
     def __init__(
-        self, index: int, initial_state=salobj.State.DISABLED, simulation_mode: int = 0
+        self,
+        index: int,
+        initial_state=salobj.State.DISABLED,
+        simulation_mode: int = 0,
     ):
         super().__init__(
             name="MTAirCompressor",
             index=index,
-            simulation_mode=simulation_mode,
             initial_state=initial_state,
+            simulation_mode=simulation_mode,
         )
 
+        self.grace_period = 600  # TODO should be configurable - DM-35280
+
         self.first_run = True
-        self.client = None
+        self.model = None
         self.simulator = None
         self.simulator_future = None
-        self.telemetry_task = utils.make_done_future()
+        self._start_by_remote = False
+        self._status_update = False
+        self._failed_time = None
+
+        self.poll_task = utils.make_done_future()
 
     @classmethod
     def add_arguments(cls, parser: argparse.ArgumentParser) -> None:
-        """Adds custom --hostname and --unit arguments."""
+        """Adds custom --hostname, --port and --unit arguments."""
         parser.add_argument(
             "--hostname",
             type=str,
             default=None,
             help="hostname. Unless specified, m1m3cam-aircomp0X.cp.lsst.org, where X is compressor index",
         )
+        parser.add_argument(
+            "--port",
+            type=int,
+            default=502,
+            help="TCP/IP port. Defaults to 502 (default Modbus TCP/IP port)",
```

Comment:
I'm a bit surprised you need this. Would you be able to get rid of all these arguments if you made the CSC configurable? If so, I suggest doing that (on another ticket).

Reply:
DM-35280.